### ISSUE TITLE
http /status issues

### ISSUE THREAD

**Issue by k5jat** [wwivbbs/wwiv#1014]
I've been playing with the new http /status. It seem the php (5.6) function `json_decode()' returns NULL.  Googling turned up that this is usually caused by either a malformed json format or there are non-standard characters in it.

You can view the results of my php at http://wwiv.fireballex.com/test.php. I'm also including my code for reference and maybe further testing.

I switched to Debian 8 for the time being. CentOS was having some non-wwiv related issues. lol

`<?php

ini_set('display_errors', 'On');

$url = 'http://wwiv.fireballex.com:81/status';

$json = file_get_contents($url);
$obj = json_decode($json,true);
echo "Echoing JSON output!<br>";
echo $json;
echo "<br>Done!<br>";

echo "Sending through jason_decode with var_dump()!<br>";
var_dump(json_decode($json, true));
echo "<br>Done!<br>";

echo "Trying to print num_instances!<br>";
print $obj->{'num_instances'};
echo "<br>Done!<br>";

?>`

**Comment by wwiv** [comment]
Thanks! I'm sure this JSON is crap that I'm spewing. Just did it quickly offhand.

I'll use a cereal to make a better json output

**Comment by wwiv** [comment]
jq isn't happy either.
```
$ curl http://localhost:8080/status | jq '.'
parse error: Objects must consist of key:value pairs at line 5, column 34
```

**Comment by k5jat** [comment]
I just wanted to help out and test it for you. I can see it being nice to have a web interface, like some of the other BBSes out there. Is this going to be a bi-directional thing? Will we be able to update the BBS from the web as well?

**Comment by wwiv** [comment]
Fixed in https://github.com/wwivbbs/wwiv/commit/0091672a9d7160beafc1b8518507220e8b9c4599

**Comment by wwiv** [comment]
initially this will just be for a UI for the socket server, read only for now. Next up will be interacting with the wwivd (shutdown, block ip, unblock IP, etc).

To do a web based BBS, this isn't the right approach for that I think (to do that, building it as a mod on an existing web platform would be better)

**Comment by granitepenguin** [comment]
waaaaayy too many new attack vectors to open up if we start allowing external access to change things.  Much better to keep this as a simple way to get info instead of being interactive

**Comment by k5jat** [comment]
The node list now starts at node #2.

**Comment by granitepenguin** [comment]
That's intentional. Node #1 is for local

**Comment by k5jat** [comment]
Ohhh ok... Looks odd. Anyway, here's my php code that works, for future reference. :)

"<?php

ini_set('display_errors', 'On');

$url = 'http://localhost:81/status';

$json = file_get_contents($url);
$obj = json_decode($json,true);

$lines = $obj['status']['lines'];

echo "Number of nodes: " . $obj['status']['num_instances'] . "<br>";
echo "Busy nodes: " . $obj['status']['used_instances'] ."<br>";

foreach ($lines as $node) {
	echo $node."<br>";
}

?>"

Note: the site is rendering my br tags. Ugh...
